sillsdev/languageforge-lexbox
# CrdtMerge API web service

Fixes #1091.

Work done:

* Sync CRDT and FW projects
* Do Send/Receive with Lexbox after sync
* Sync CRDT and FW projects again if S/R pulled down any changes from others
* Create API endpoint to trigger all of this

Still remaining:

* <del>Create k8s deployment</del> Put off for later PR
* <del>Write tests to exercise it by sending API requests and verifying correct results happen</del> Put off for later PR

## To test

* Edit appsettings.Development.json and choose a different folder than `../../hgweb/repos` if you want, otherwise the test will put files into `lexbox/hgweb/repos`
* Run `dotnet run` inside the CrdtMerge folder
* Send a POST request to http://localhost:5275/sync?projectCode=sena-3
* Should take a minute or so to complete, watch the `dotnet run` logs while you wait
* Verify that the JSON result has the right number of changes (should be 1462, one per entry)
* Verify that the CRDT SQLite files showed up in `lexbox/hgweb/repos`, or wherever you put them
